src/core/symbology-ng/qgspointdisplacementrenderer.cpp

```diff
@@ -301,6 +308,69 @@ void QgsPointDisplacementRenderer::calculateSymbolAndLabelPositions( QgsSymbolRe
       }
       break;
     }
+    case Grid:
+    {
+      double centerDiagonal = symbolContext.renderContext().convertToPainterUnits( M_SQRT2 * mCenterSymbol->size(),
+                              mCenterSymbol->sizeUnit(), mCenterSymbol->sizeMapUnitScale() );
+      int pointsRemaining = nPosition;
+      gridSize = ceil( sqrt( pointsRemaining ) );
+      if ( pointsRemaining - pow( gridSize - 1, 2 ) < gridSize )
+        gridSize -= 1;
+      double pointRadius = ( centerDiagonal / 2.0 + symbolDiagonal / 2.0 ) + symbolDiagonal;
+
+      int yIndex = 0;
+      while ( pointsRemaining > 0 )
+      {
+        for ( int xIndex = 0; xIndex < gridSize; ++xIndex )
```

Comment:
for ( int xIndex = 0; xIndex < gridSize && pointsRemaining > 0; ++xIndex )

otherwise this inner loop runs more times than is necessary